include/caffe/neuron_layers.hpp

```diff
@@ -654,6 +654,89 @@ class ThresholdLayer : public NeuronLayer<Dtype> {
   Dtype threshold_;
 };
 
+/**
+ * @brief Parameterized Rectified Linear Unit non-linearity @f$
+ *        y_i = \max(0, x_i) + a_i \min(0, x_i)
+ *        @f$. The differences from ReLULayer are 1) negative slopes are
+ *        learnable though backprop and 2) negative slopes can vary across
+ *        channels.
+ */
+template <typename Dtype>
+class PReLULayer : public NeuronLayer<Dtype> {
+ public:
+  /**
+   * @param param provides PReLUParameter prelu_param,
+   *     with PReLULayer options:
+   *   - init_value (\b optional, default 0.25).
+   *     all negative slopes over channels are set to this value.
```

Comment:
How about using a `Filler` for this (like `InnerProductLayer` and `ConvolutionLayer`)?